sound/soc/sof/intel/hda-dai.c

```diff
@@ -372,7 +372,10 @@ static int hda_link_hw_free(struct snd_pcm_substream *substream,
 	if (ret < 0)
 		return ret;
 
-	snd_hdac_ext_stream_decouple(bus, link_dev, false);
+	/* couple host and link DMA if host DMA channel is idle */
+	if (!hstream->opened)
+		snd_hdac_ext_stream_decouple(bus, link_dev, false);
+
```

Comment:
we actually don't need this part as it will be done in line 385 below?
```
snd_hdac_ext_stream_release()
```

Reply:
@keyonjie good point. removed now